File: Easy/pianostudioattivform/meta_pianostudioattivformseganagstuview/meta_pianostudioattivformseganagstuview.cs
/*
    Easy
    Copyright (C) 2020 Università degli Studi di Catania (www.unict.it)
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

﻿using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_pianostudioattivformseganagstuview
{
    public class Meta_pianostudioattivformseganagstuview : Meta_easydata 
	{
        public Meta_pianostudioattivformseganagstuview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "pianostudioattivformseganagstuview") {
				Name = "Attività formative pianificate";
			EditTypes.Add("seganagstu");
			ListingTypes.Add("seganagstu");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"idcorsostudio","iddidprog","idiscrizione","idpianostudio","idpianostudioattivform","idreg"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "seganagstu": {
						DescribeAColumn(T, "anno", "Anno di corso", nPos++);
						DescribeAColumn(T, "attivform_title", "Attività formativa del corso", nPos++);
						DescribeAColumn(T, "attivform_scelta_title", "Attività formativa che lo studente svolgerà", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		//$GetSortings$

		//$GetStaticFilter$
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/pianostudioattivform/meta_pianostudioattivformseganagstuview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_pianostudioattivformseganagstuview() {
        MetaData.apply(this, ["pianostudioattivformseganagstuview"]);
        this.name = 'meta_pianostudioattivformseganagstuview';
    }

    meta_pianostudioattivformseganagstuview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_pianostudioattivformseganagstuview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'seganagstu':
						this.describeAColumn(table, 'anno', 'Anno di corso', null, 20, null);
						this.describeAColumn(table, 'attivform_title', 'Attività formativa del corso', null, 30, -1);
						this.describeAColumn(table, 'attivform_scelta_title', 'Attività formativa che lo studente svolgerà', null, 40, -1);
//$objCalcFieldConfig_seganagstu$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["idreg", "iddidprog", "idiscrizione", "idcorsostudio", "idpianostudio", "idpianostudioattivform"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			//$getSorting$

        });

    window.appMeta.addMeta('pianostudioattivformseganagstuview', new meta_pianostudioattivformseganagstuview('pianostudioattivformseganagstuview'));

	}());
